### ISSUE TITLE
MongoCursor hasNext iteration problem if nest find ，cursor will not iteration

### ISSUE THREAD

**Issue by shanghaiyangming** [alcaeus/mongo-php-adapter#91]
```
<?php
include "vendor/autoload.php";

$client = new MongoClient('mongodb://127.0.0.1:27017');
$test1 = $client->selectCollection('test', 'test1');
$test2 = $client->selectCollection('test', 'test2');
$test1->drop();
$test2->drop();

for ($i = 0; $i < 5; $i ++) {
    $inertData = array(
        'a' => $i,
        'b' => time()
    );
    var_dump($test1->insert($inertData));
    $inertData1 = array(
        'a' => $i,
        'b' => time()
    );
    var_dump($test2->insert($inertData1));
}

$cursor = $test1->find([],['a'=>true]);
$cursor->limit(10);
$cursor->sort(array(
    '_id' => - 1
));

$data = array();
while ($cursor->hasNext()) {
    $row = $cursor->getNext();
    $test2->count(array(
        'a' => $row['a']
    ));
    $data[] = $row;
}

var_dump($data);
```

**# the correct result  is a from 0 to 4

but the real result is :  
double 3 like this:**
```
array(5) {
  [0]=>
  array(2) {
    ["_id"]=>
    object(MongoId)#43 (1) {
      ["objectID":"MongoId":private]=>
      object(MongoDB\BSON\ObjectID)#36 (1) {
        ["oid"]=>
        string(24) "5704d1c991a65e2384002817"
      }
    }
    ["a"]=>
    int(3)
  }
  [1]=>
  array(2) {
    ["_id"]=>
    object(MongoId)#48 (1) {
      ["objectID":"MongoId":private]=>
      object(MongoDB\BSON\ObjectID)#47 (1) {
        ["oid"]=>
        string(24) "5704d1c991a65e2384002817"
      }
    }
    ["a"]=>
    int(3)
  }
  [2]=>
  array(2) {
    ["_id"]=>
    object(MongoId)#45 (1) {
      ["objectID":"MongoId":private]=>
      object(MongoDB\BSON\ObjectID)#44 (1) {
        ["oid"]=>
        string(24) "5704d1c991a65e2384002815"
      }
    }
    ["a"]=>
    int(2)
  }
  [3]=>
  array(2) {
    ["_id"]=>
    object(MongoId)#52 (1) {
      ["objectID":"MongoId":private]=>
      object(MongoDB\BSON\ObjectID)#51 (1) {
        ["oid"]=>
        string(24) "5704d1c991a65e2384002813"
      }
    }
    ["a"]=>
    int(1)
  }
  [4]=>
  array(2) {
    ["_id"]=>
    object(MongoId)#54 (1) {
      ["objectID":"MongoId":private]=>
      object(MongoDB\BSON\ObjectID)#53 (1) {
        ["oid"]=>
        string(24) "5704d1c991a65e2384002811"
      }
    }
    ["a"]=>
    int(0)
  }
}

```

**Comment by shanghaiyangming** [comment]
All hasNext() getNext() will triggering this bug.

**Comment by alcaeus** [comment]
Thanks for the example! This will be fixed in 1.0.2.
